drivers: ethernet: fix coverity issue

gudnimg opened the pull request:
If return value is negative (failed) then log the error with a
message.

Coverity-CID: 219519
Fixes #32927

Signed-off-by: Guðni Már Gilbert <gudni.m.g@gmail.com>

jukkar commented on drivers/ethernet/eth_sam_gmac.c:
```diff
@@ -1792,9 +1792,14 @@ static void get_mac_addr_from_i2c_eeprom(uint8_t mac_addr[6])
-	i2c_write_read(dev, CONFIG_ETH_SAM_GMAC_MAC_I2C_SLAVE_ADDRESS,
-		       &iaddr, CONFIG_ETH_SAM_GMAC_MAC_I2C_INT_ADDRESS_SIZE,
-		       mac_addr, 6);
+	int ret = i2c_write_read(dev, CONFIG_ETH_SAM_GMAC_MAC_I2C_SLAVE_ADDRESS,
```
We try to introduce variables at the beginning of the block, so could you place the `int ret;` at the beginning of the function. Other than that, LGTM

GunZi200 replied:
Fixed 😄

GunZi200 replied:
@jukkar I wasn't sure if the order of the variables at the top of the function mattered. But I put `int ret` below `uint32_t`. I vaguely remember seeing this being an issue with some standard like MISRA but not sure.

jukkar commented on drivers/ethernet/eth_sam_gmac.c:
```diff
@@ -1792,9 +1792,14 @@ static void get_mac_addr_from_i2c_eeprom(uint8_t mac_addr[6])
 		return;
```
Almost forgot, please change the commit subject to `drivers: ethernet: gmac: Fix coverity issue`

GunZi200 replied:
Fixed 😄

jukkar reviewed (approved):
Thanks!

pfalcon commented on drivers/ethernet/eth_sam_gmac.c:
```diff
@@ -1785,16 +1785,22 @@ static void get_mac_addr_from_i2c_eeprom(uint8_t mac_addr[6])
+			   mac_addr, 6);
+
+	if (ret != 0) {
+		LOG_ERR("I2C: i2c_write_read() failed");
```
I wanted to give this comment with previous PR (regarding samples/*), but there it's not so important. But this driver can be used in production code, and if log an error at, let's make it be helpful for user/engineer to understand what's the problem. Something like "failed to read MAC addr" would ok (and that's optimized for code size, "failed to read MAC addr from EEPROM" would be better, but too long).

GunZi200 replied:
@pfalcon I agree. I made the change. Let me know if you'd like to drop the `"I2C: "` part.

gudnimg committed:
drivers: ethernet: gmac: fix coverity issue

If return value is negative (failed) then log the error with a
message.

Coverity-CID: 219519
Fixes #32927

Signed-off-by: Guðni Már Gilbert <gudni.m.g@gmail.com>
